File: apps/codemap-search/tests/fixtures/extract/sample.php
<main>mixed html</main>
<?php
namespace Demo;

use Demo\Support\Clock as AppClock;
require_once './worker.php';

#[Deprecated]
class Worker
{
    public const STATE = 'ready';
    public string $status = "idle";

    public function run(AppClock $clock): void
    {
        helper();
        $clock->tick();
        AppClock::now();
        $child = new Worker();
    }
}

function helper(): void {}

enum Mode
{
    case Fast;
}

File: apps/codemap-search/tests/fixtures/extract/sample.rb
require_relative "worker"

module Demo
  class Worker
    attr_reader :status, :name
    STATE = "ready"

    def run(clock)
      helper()
      clock.tick()
    end

    private

    def helper
    end
  end
end

def top_level
  Demo::Worker.new
end

